File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/Archiwum/2020w/ID07IO1AB/Builder/src/IdBuilder.java

import java.util.ArrayList;

public class IdBuilder
{
   private final ArrayList<Long> list=new ArrayList<>();
   private String IfEmpty="null";
   private String Separator=", ";
   public IdBuilder clear() { list.clear(); return this; }
   public IdBuilder append(long id) { list.add(id); return this; }
   public IdBuilder setIfEmpty(String IfEmpty) { this.IfEmpty=IfEmpty; return this; }
   public IdBuilder setSeparator(String Separator) { this.Separator=Separator; return this; }

   @Override public String toString()
   {
      if(list.isEmpty()) return IfEmpty;
      list.sort((long a,long b) -> { return a<b; });
      StringBuilder sb=new StringBuilder();
      /*
      for(int i=0;i<list.size();++i)
      {
         if(i>0) sb.append(Separator);
         sb.append(list.get(i));
      }
      */
      for(int i=0;i<list.size();sb.append(list.get(i++))) if(i>0) sb.append(Separator);
      /*
      boolean first=true;
      for(long Id:list)
      {
         if(first) first=false;
         else sb.append(Separator);
         sb.append(Id);
      }
      */
      return sb.toString();
   }
}

File: semestr-8/Jezyk C#/alexander/Csharp/Archiwum/2021s/Id06IO1/Builder/Builder/IdBuilder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Builder
{
    class IdBuilder
    {
        private List<long> list=new List<long>();
        private string Delimetr=", ";
        private string IfEmpty="null";

        // ... where id in (1, 13, 666)
        // ... where id in (null)
        // ib.Append(1).Append(13).SetDelimetr(";").ToString();
        public IdBuilder Append(long id)
        {
            list.Add(id);
            return this;
        }

        public IdBuilder Clear()
        {
            list.Clear();
            return this;
        }

        public IdBuilder SetDelimetr(string Delimetr)
        {
            this.Delimetr=Delimetr;
            return this;
        }

        public IdBuilder SetIfEmpty(string IfEmpty)
        {
            this.IfEmpty=IfEmpty;
            return this;
        }

        public override string ToString()
        {
            if (list.Count <= 0) return IfEmpty;
            StringBuilder sb=new StringBuilder();
            bool first=true;
            foreach(long id in list)
            {
                if(first) first=false; else sb.Append(Delimetr);
                sb.Append(id);
            }
            return sb.ToString();
        }
    }
}
